moped-editor/src/views/projects/projectView/NoteInputQuill.js

```diff
@@ -47,6 +52,23 @@ const CommentInputQuill = ({
     ref?.current.focus();
   }, []);
 
+  /**
+   * Defines the NewNoteTypeButton with a toggle style-change behavior.
+   * @param {Object} props
+   * @return {JSX.Element}
+   * @constructor
+   */
+  const NewNoteTypeButton = (props) => (
```

Comment:
You can move this component declaration outside of the `NoteInputQuill` component. I think it's fine to leave the component in the same file—IMO it's a judgement call when it makes sense to break components into separate files.

Reply:
yeah, agree here. If this definition stays inside the `NoteInputQuill` component function, it is recreated on every render. Definitely a legacy anti-pattern that we've been slowly chipping away at in the codebase. 🧹

Reply:
just saw that this is updated now! 🚀